src/UserItem.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import styled from 'styled-components';
+
+const Container = styled.div`
+    display: flex;
+    flex-direction: column;
+    margin: 2px;
+    text-align: center;
+`
+const StyledImage = styled.img`
+    height: 50px;
+    width: 50px;
+    border-radius: 5px;
+`;
+
+const StyledButton = styled.button`
+    border: none;
+    background: none;
+`
+
+type UserItemProps = {
+    onSubmit: (form:number) => void;
+    id: number;
+    name: string;
+    src: string;
+};
+
+function UserItem({onSubmit,id,name,src}:UserItemProps){
+    
+    // user 바꾸기
+    const handleClick = (e:any) => {
+        onSubmit(e.target.value);
+    }
```

Comment:
```suggestion
  const handleClick = (e: React.MouseEvent<HTMLButtonElement>) => {
    onSubmit(Number(e.currentTarget.value));
  };
```
이벤트 객체의 타입도 이렇게 정의해줄 수 있습니다. `MouseEvent`에는 `e.target` 속성이 없네요. 대신 `currentTarget`을 사용하는 경우가 많이 있다고 합니다. 
value 타입이 들어갈때는 number로 들어가는데, `e.currentTarget.value`로 뽑아서 쓸때는 string으로 나오네요??훔.. 좋은 방법인지는 저도 조금 더 고민해봐야될 것 같습니다ㅎㅎ
[target 과 currentTarget 차이](https://seungddak.tistory.com/130)